tests/SoapSerializationTest.php

```diff
@@ -38,6 +39,36 @@ public function testSerialization(string $name, int $complexity, DateTime $date)
         static::assertSame($date->format('Y-m-d'), strval($message->since));
     }
 
+    /**
+     * @dataProvider provideLanguage
+     *
+     * @param string $name
+     * @param int $complexity
+     * @param DateTime $date
+     */
+    public function testSerializationWithPrefix(string $name, int $complexity, DateTime $date)
```

Comment:
This tests the registered namespace and prefix, not is the root node for the message actual contains that prefix.
I think you should use DOMDocument here to ensure the prefix is set on the expected element(s)

(see SoapHeaderEventSubscriberTest.php)